src/App.tsx

```diff
@@ -1,222 +1,190 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import cn from 'classnames';
+import React, { useContext, useEffect, useMemo } from 'react';
+import { Route, Routes } from 'react-router-dom';
+
+import { deleteTodo, getTodos } from './api/todos';
+import { Error } from './types';
+
+import { AppContext } from './components/AppProvider/AppProvider';
+import { Filter } from './components/Filter';
+import { NewTodo } from './components/NewTodo';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const {
+    userId,
+    todos,
+    error,
+    setTodos,
+    setError,
+    setIsLoadingMany,
+    setIsDeleting,
+  } = useContext(AppContext);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
-    }
+    // eslint-disable-next-line func-names
+    (async function () {
+      setTodos(await getTodos(userId || 0));
+    }());
   }, []);
```

Comment:
You can create a async function outside of useEffect then call it in the useEffect